### PR TITLE
Fix shellcheck failures of hack/update-gofmt.sh hack/update-translations.sh

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Fix shellcheck failures of hack/update-gofmt.sh hack/update-translations.sh

**Which issue(s) this PR fixes**:

Ref #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/update-gofmt.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
```
Comment (SataQiu) [inline hack/update-gofmt.sh]: change reason:
```
In hack/update-gofmt.sh line 23:
KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
                     ^-- SC2128: Expanding an array without an index only gives the first element.
```

**Review by fejta — APPROVED** [review]
/lgtm
/approve

**Comment by SataQiu** [comment]
@fejta 
I have resubmitted the code to resolve some conflicts.

File: hack/update-translations.sh [hunk 2]
```diff
@@ -61,7 +61,7 @@ fi
 
 if [[ "${generate_pot}" == "true" ]]; then
   echo "Extracting strings to POT"
-  go-xgettext -k=i18n.T ${KUBECTL_FILES} > tmp.pot
+  go-xgettext -k=i18n.T "${KUBECTL_FILES}" > tmp.pot
```
Comment (fejta) [inline hack/update-translations.sh]: Any chance this will cause problems?

**Review by fejta — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**